src/test/java/org/breedinginsight/brapps/importer/GermplasmTemplateMap.java

```diff
@@ -0,0 +1,615 @@
+        assertEquals(HttpStatus.OK, response.getStatus());
+        JsonObject result = JsonParser.parseString(response.body()).getAsJsonObject().getAsJsonObject("result");
+        String importId = result.get("importId").getAsString();
+
+        // Process data
+        String url = String.format("/programs/%s/import/mappings/%s/data/%s/%s", validProgram.getId(), germplasmImportId, importId, commit ? "commit" : "preview");
+        Flowable<HttpResponse<String>> processCall = client.exchange(
+                PUT(url, new HashMap<>())
+                        .cookie(new NettyCookie("phylo-token", "test-registered-user")), String.class
+        );
+        return processCall;
+    }
+
+    public HttpResponse<String> getUploadedFile(String importId) throws InterruptedException {
+        Flowable<HttpResponse<String>> call = client.exchange(
+                GET(String.format("/programs/%s/import/mappings/%s/data/%s?mapping=true", validProgram.getId(), germplasmImportId, importId))
+                        .contentType(MediaType.MULTIPART_FORM_DATA_TYPE)
+                        .cookie(new NettyCookie("phylo-token", "test-registered-user")), String.class
+        );
+        HttpResponse<String> response = call.blockingFirst();
+
+        if (response.getStatus().equals(HttpStatus.ACCEPTED)) {
+            Thread.sleep(1000);
+            return getUploadedFile(importId);
+        } else {
+            return response;
+        }
+
+
+    }
+
+
+    public void checkBasicResponse(JsonObject germplasm, Table fileData, Integer i) {
+
+        // Germplasm display name
+        assertEquals(fileData.getString(i, "Name"), germplasm.get("defaultDisplayName").getAsString(), "Wrong display name");
+        // Entry Number
+        JsonObject additionalInfo = germplasm.getAsJsonObject("additionalInfo");
+        assertEquals(fileData.getString(i, "Entry No"), additionalInfo.get("importEntryNumber").getAsString(), "Wrong entry number");
+        // Created By User ID
+        assertEquals(testUser.getId().toString(), additionalInfo.getAsJsonObject("createdBy").get("userId").getAsString(), "Wrong createdBy userId");
+        // Created by User name
+        assertEquals(testUser.getName(), additionalInfo.getAsJsonObject("createdBy").get("userName").getAsString(), "Wrong createdBy userId");
+        // Breeding Method
+        String breedingMethodName = fileData.getString(i, "Breeding Method");
+        BreedingMethodEntity breedingMethod = breedingMethodDAO.findByNameOrAbbreviation(breedingMethodName).get(0);
+        assertEquals(breedingMethod.getId().toString(), additionalInfo.get("breedingMethodId").getAsString(), "Wrong Breeding Method ID");
+        assertEquals(breedingMethod.getName(), additionalInfo.get("breedingMethod").getAsString(), "Wrong Breeding Method name");
+        // Seed source
+        assertEquals(fileData.getString(i, "Source"), germplasm.get("seedSource").getAsString(), "Wrong seed source description");
```

Comment:
Should the failure message change now that we switched from seed source description to seed source?

Reply:
updated